src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +129,92 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
-        ???
+        Set up the formatting for the scheduler batch script.
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm and none.
+        slurmqueues = ["hera", "jet", "orion", "stampede"]
+        if platform["queue"] is None and platform["machine"] in slurmqueues:
+            platform["queue"] = "slurm"
+        else:
+            platform["queue"] = "none"
+
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": platform["queue"],
+            "scheduler": platform["scheduler"],
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self, log: Optional[Logger] = None) -> None:
         """
         Runs FV3.
         """
+        # Set up logging.
+        if log is None:
+            name = f"{inspect.stack()[0][3]}"
+            log = cli_helpers.setup_logging(
+                log_file="/dev/stdout",
+                log_name=name,
+                quiet=False,
+                verbose=False,
+            )
+        # Read in the config file.
+        forecast_config = config.YAMLConfig(self._config_file)
+
+        # Define forecast model.
+        forecast_model = forecast_config["forecast"]["EXTRN_MDL_NAME"]
+        machine = forecast_config["platform"]["machine"]
+
+        # Prepare directories.
+        run_directory = forecast_config["forecast"]["FCSTDIR"]
+        self.create_directory_structure(run_directory, "delete")
+
+        static_files = forecast_config["forecast"]["STATIC"]
+        self.stage_files(run_directory, static_files, link_files=False)
+        cycledep_files = forecast_config["forecast"]["CYCLEDEP"]
+        self.stage_files(run_directory, cycledep_files, link_files=True)
+
+        # Create the job script.
+        platform = {key: forecast_config["platform"][key] for key in forecast_config["platform"]}
+        platform_resources = self.resources(platform)
+        batch_script = self.batch_script(platform_resources)
+        args = "--export=ALL -n"
+        run_command = self.run_cmd(
+            args,
+            run_cmd=forecast_config["forecast"]["run_cmd"],
+            exec_name=forecast_config["forecast"]["exec_name"],
+        )
+
+        if self._dry_run:
+            # Apply switch to allow user to view the run command of config.
+            # This will not run the job.
+            log.info(f"Configuration: {forecast_model} {machine}")
+            log.info(f"Run command: {run_command} {batch_script}")
+            return
+
+        # Run the job.
+        if self._outfile is not None:
+            outpath = run_directory.join(self._outfile)
+            with open(outpath, "w", encoding="utf-8") as f:
+                subprocess.run(
+                    f"{run_command} {batch_script}",
```

Comment:
This subprocess should run `f"sbatch {batch_script}"`. The batch command should come from a property in scheduler, which I don't think currently exists. 

@maddenp-noaa Any ideas on where to get this piece of info?

Reply:
There are `Slurm` and `PBS` classes in `scheduler.py`: These facts about the respective batch managers could be class attributes or static/class methods (the latter would allow us to declare them abstract in the `JobScheduler` base class to ensure that they are implemented in subclasses); or maybe (abstract in `JobScheduler`, concrete in `Slurm` / `PBS`) instance methods since IIRC we can get an instance in `FV3Forecast` (it looks like we already get, and then discard, and instance in the statement `return JobScheduler.get_scheduler(job_resources).batch_script`, so we could obtain it, cache it, and call it for any facts about the batch manager we with e.g. a method on `FV3Forecast` like
```
from functools import cached_property
...
@cached_property
def batch_manager(self):
    return JobScheduler.get_scheduler(self.config[...])
```
That assumes that, as @christinaholtNOAA suggested, `self.config` is defined in the constructor, by instantiating a `YAMLConfig` object.

(It may not be much of / any optimization to cache this value, but it also shouldn't be _necessary_ to obtain it more than once, I think.)

Then, elsewhere in the class, code could call methods on this object like `self.batch_manager.batch_script` (already exists) or `self.batch_manager.submit_command` (doesn't exist yet).

Reply:
Upon reflection, we probably don't want both "batch manager" and "job scheduler" terminology and should pick one or the other. "Since "job scheduler" is already in the codebase, maybe we should stick with that.

Reply:
Hm - for defining in the constructor, I've moved  `self.config_data = config.YAMLConfig(self._config_file)` into `Driver.__init__` which makes it more broadly accessible. This does annoyingly require me to rework all the tests that had been getting away with `fcstobj = FV3Forecast(config_file="/not/used")` - working on it!